### PR TITLE
p2p: remove term "whitelist"

### PR DESCRIPTION
**Pull request by baptiste-b-pegasys**
split of #23290
only p2p package, including the review of @fjl

### CODE REVIEW

File: p2p/discover/v4_udp.go [hunk 1]
```diff
@@ -583,7 +583,7 @@ func (t *UDPv4) nodeFromRPC(sender *net.UDPAddr, rn v4wire.Node) (*node, error)
 		return nil, err
 	}
 	if t.netrestrict != nil && !t.netrestrict.Contains(rn.IP) {
-		return nil, errors.New("not contained in netrestrict whitelist")
+		return nil, errors.New("not contained in netrestrict allow list")
```
Comment (baptiste-b-pegasys) [inline p2p/discover/v4_udp.go:586]:
```suggestion
		return nil, errors.New("not contained in netrestrict list")
```

File: p2p/dial.go [hunk 2]
```diff
@@ -73,12 +73,12 @@ func nodeAddr(n *enode.Node) net.Addr {
...
-	errAlreadyDialing   = errors.New("already dialing")
-	errAlreadyConnected = errors.New("already connected")
-	errRecentlyDialed   = errors.New("recently dialed")
-	errNotWhitelisted   = errors.New("not contained in netrestrict whitelist")
-	errNoPort           = errors.New("node does not provide TCP port")
+	errSelf                 = errors.New("is self")
+	errAlreadyDialing       = errors.New("already dialing")
+	errAlreadyConnected     = errors.New("already connected")
+	errRecentlyDialed       = errors.New("recently dialed")
+	errNotNetRestrictListed = errors.New("not contained in netrestrict list")
```
Comment (fjl) [inline p2p/dial.go:80]: You can really call it `errNetRestrict`, it's an internal error.

File: p2p/discover/common.go [hunk 3]
```diff
@@ -41,7 +41,7 @@ type Config struct {
 	PrivateKey *ecdsa.PrivateKey
 
 	// These settings are optional:
-	NetRestrict  *netutil.Netlist   // network whitelist
+	NetRestrict  *netutil.Netlist   // network restriction list
```
Comment (fjl) [inline p2p/discover/common.go:44]: This comment could be better. It's the list that nodes *are restricted to*. How about

```
// list of allowed IP networks
```

File: p2p/server.go [hunk 4]
```diff
@@ -903,7 +903,7 @@ func (srv *Server) checkInboundConn(remoteIP net.IP) error {
 	}
 	// Reject connections that do not match NetRestrict.
 	if srv.NetRestrict != nil && !srv.NetRestrict.Contains(remoteIP) {
-		return fmt.Errorf("not whitelisted in NetRestrict")
+		return fmt.Errorf("not in the NetRestrict list")
```
Comment (fjl) [inline p2p/server.go:906]: "not in netrestrict list"

**Review by fjl — APPROVED** [review]

**Pull request merged**